drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,476 @@
+
+    uint8_t userreg;
+    uint8_t userreg2;
+    i2c_result = sht2x_read_userreg(dev, &userreg);
+    if (i2c_result != SHT2X_OK) {
+        return i2c_result;
+    }
+    DEBUG("[SHT2x] User Register=%02x\n", userreg);
+
+    if ((userreg & SHT2X_USER_RESOLUTION_MASK) != (uint8_t)dev->params.resolution) {
+        userreg &= ~SHT2X_USER_RESOLUTION_MASK;
+        userreg |= (uint8_t)dev->params.resolution;
+        i2c_result = sht2x_write_userreg(dev, userreg);
+        if (i2c_result != SHT2X_OK) {
+            return i2c_result;
+        }
+        i2c_result = sht2x_read_userreg(dev, &userreg2);
+        if (i2c_result != SHT2X_OK) {
+            return i2c_result;
+        }
+        DEBUG("[SHT2x] New User Register=%02x\n", userreg2);
+        if (userreg != userreg2) {
+            return SHT2X_ERR_OTHER;
+        }
+    }
+
+    return SHT2X_OK;
+}
+
+int sht2x_reset(sht2x_t* dev)
+{
+    int i2c_result;
+    cmd_t command = eSoftResetCmd;
+
+    /* Acquire exclusive access */
+    i2c_acquire(BUS);
+
+    DEBUG("[SHT2x] write command: addr=%02x cmd=%02x\n", ADDR, (uint8_t)command);
+    i2c_result = i2c_write_byte(BUS, ADDR, (uint8_t)command, 0);
+
+    i2c_release(BUS);
+
+    if (i2c_result != 0) {
+        return SHT2X_ERR_I2C;
+    }
+
+    return SHT2X_OK;
+}
+
+/*
+ * Returns temperature in centi DegC.
+ */
+int16_t sht2x_read_temperature(const sht2x_t* dev)
+{
+    uint16_t raw_value;
+    int i2c_result;
+    if (dev->params.measure_mode == SHT2X_MEASURE_MODE_NO_HOLD) {
+        i2c_result = read_sensor_poll(dev, eTempNoHoldCmd, &raw_value);
+    } else {
+        i2c_result = read_sensor(dev, eTempHoldCmd, &raw_value);
+    }
+    if (i2c_result != SHT2X_OK) {
+        return INT16_MIN;
+    }
+    return (-46.85 + 175.72 / 65536.0 * raw_value) * 100;
+}
+
+/*
+ * Returns humidity in centi %RH (i.e. the percentage times 100).
+ */
+uint16_t sht2x_read_humidity(const sht2x_t *dev)
+{
+    uint16_t raw_value;
+    int i2c_result;
+    if (dev->params.measure_mode == SHT2X_MEASURE_MODE_NO_HOLD) {
+        i2c_result = read_sensor_poll(dev, eRHumidityNoHoldCmd, &raw_value);
+    } else {
+        i2c_result = read_sensor(dev, eRHumidityHoldCmd, &raw_value);
+    }
+    if (i2c_result != SHT2X_OK) {
+        return 0;
+    }
+    return 100 * (-6.0 + 125.0 / 65536.0 * raw_value);
+}
+
+static int _sht2x_add_ident_byte(uint8_t * buffer, size_t buflen, uint8_t b, size_t ix)
+{
+    if (ix < buflen) {
+        buffer[ix++] = b;
+    }
+    return ix;
+}
+
+int sht2x_read_ident(const sht2x_t *dev, uint8_t * buffer, size_t buflen)
+{
+    uint8_t data1[8];        /* SNB_3, CRC, SNB_2, CRC, SNB_1, CRC, SNB_0, CRC */
+    uint8_t data2[6];        /* SNC_1, SNC_0, CRC, SNA_1, SNA_0, CRC */
+    size_t ix;
+    int res;
+
+    i2c_acquire(BUS);
+    res = i2c_read_regs(BUS, ADDR,
+                        first_mem_addr, data1, sizeof(data1), I2C_REG16);
+    i2c_release(BUS);
+    if (res < 0) {
+        return -1;
```

Comment:
return enum val?

Reply:
OK I'll do a `return res` and document it